nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -442,6 +444,13 @@ let
         proxy_set_header Upgrade $http_upgrade;
         proxy_set_header Connection $connection_upgrade;
       ''}
+      ${optionalString (config.uwsgiPass != null && !cfg.proxyResolveWhileRunning)
+        "uwsgi_pass ${config.uwsgiPass};"
+      }
+      ${optionalString (config.uwsgiPass != null && cfg.proxyResolveWhileRunning) ''
```

Comment:
I'm a little bit torn on this.

On one hand I think it's surprising that the `proxy*` options have an effect on uwsgi. OTOH it seems tedious to add counterparts for the uwsgi case.

I'm slightly leaning towards adding uwsgi counterparts for everything we need here. What do others think of this?

Reply:
I've added the ones that existed. uwsgi has no http_version or set_header commands.

Reply:
I'm also leaning towards adding uwsgi counterparts because this behavior is not intuitive. If we keep the proposed behavior in this PR we IMHO should at least mention in the proxy-related options documentation that they also apply to uwsgi.

As far as I can see this only applies to:
  * `recommendedProxySettings`
  * `proxyResolveWhileRunning`
  * `proxyTimeout`

Reply:
Just done that :)